agentlace/zmq_wrapper/req_rep.py

```diff
@@ -33,19 +33,134 @@ def __init__(self,
         logging.basicConfig(level=log_level)
         logging.debug(f"Req-rep server is listening on port {port}")
 
+    def reset_socket(self):
+        """Reset the ZMQ socket if it's in a bad state"""
+        try:
+            if self.socket:
+                try:
+                    # Try to unbind first if possible
+                    self.socket.unbind(f"tcp://*:{self.port}")
+                except zmq.ZMQError:
+                    pass  # Socket might not be bound
+                self.socket.close()
+                self.socket = None
+                time.sleep(0.1)  # Give OS time to free up the port
+
+            # Create new socket
+            self.socket = self.context.socket(zmq.REP)
+            self.socket.setsockopt(zmq.LINGER, 0)
+            self.socket.setsockopt(zmq.RCVTIMEO, 1000)  # 1 second timeout
```

Comment:
Can we use a self._timeout arg, here. it can share with `poll_timeout` in run().

Then this can add a arg in `ReqRepServer.init(...., timeout_ms)`